hera_cal/apply_cal.py

```diff
@@ -227,17 +246,44 @@ def apply_cal(data_infilename, data_outfilename, new_calibration, old_calibratio
         add_to_history += '\nOLD_CALFITS_HISTORY: ' + old_hc.history + '\n'
     else:
         old_gains, old_flags = None, None
-
+    hd = io.HERAData(data_infilename, filetype=filetype_in)
     add_to_history = version.history_string(add_to_history)
-
+    no_red_weights = redundant_weights is None
     # partial loading and writing using uvh5
     if nbl_per_load is not None:
         if not ((filetype_in == 'uvh5') and (filetype_out == 'uvh5')):
             raise NotImplementedError('Partial writing is not implemented for non-uvh5 I/O.')
-        hd = io.HERAData(data_infilename, filetype='uvh5')
         for attribute, value in kwargs.items():
             hd.__setattr__(attribute, value)
-        for data, data_flags, _ in hd.iterate_over_bls(Nbls=nbl_per_load):
+        all_reds = redcal.get_reds(hd.antpos, pols=hd.pols, bl_error_tol=bl_error_tol, include_autos=True)
+        if redundant_average:
+            # initialize a redunantly averaged HERAData on disk
+            # first copy the original HERAData
+            all_red_antpairs = [[bl[:2] for bl in grp] for grp in all_reds if grp[-1][-1] == hd.pols[0]]
+            hd_red = io.HERAData(data_infilename)
+            # go through all redundant groups and remove the groups that do not
+            # have baselines in the data. Each group is still labeled by the
+            # first baseline of each group regardless if that baseline is in
+            # the data file.
+            red_data_bls = []
+            for grpnum, grp in enumerate(all_red_antpairs):
+                nbls = 0
+                data_grp = []
+                for bl in grp:
+                    if bl in hd.antpairs or bl[:-1] in hd.antpairs:
+                        nbls += 1
```

Comment:
Seems like this can be done easier with a `break` statement, rather than accumulating more variables you don't need like `data_grp` and `nbls`... like if `np.any([bl in hd.antpairs for bl in grp]) or np.any([bl[::-1] in hd.antpairs for bl in grp])`

Reply:
Now that I know about `filter_reds`, this is a one-liner.